collect_app/src/main/java/org/odk/collect/android/utilities/FileUtils.java

```diff
@@ -163,127 +138,6 @@ private static String actualCopy(File sourceFile, File destFile) {
-     * definition does not contain any field of type geopoint.
-     *
-     * The first geopoint is either of:
-     *      (1) the first geopoint in the body that is not in a repeat
-     *      (2) if the form has a setgeopoint action, the first geopoint in the instance that occurs
-     *          before (1) or (1) if there is no geopoint defined before it in the instance.
-     */
-    private static String getOverallFirstGeoPoint(FormDef formDef) {
-        TreeReference firstTopLevelBodyGeoPoint = getFirstToplevelBodyGeoPoint(formDef);
-
-        if (!formDef.hasAction(SetGeopointActionHandler.ELEMENT_NAME)) {
-            return firstTopLevelBodyGeoPoint == null ? null : firstTopLevelBodyGeoPoint.toString(false);
-        } else {
-            return getInstanceGeoPointBefore(firstTopLevelBodyGeoPoint, formDef.getMainInstance().getRoot());
-        }
-    }
-
-    /**
-     * Returns the reference of the first geopoint in the body that is not in a repeat.
-     */
-    private static TreeReference getFirstToplevelBodyGeoPoint(FormDef formDef) {
-        if (formDef.getChildren().size() == 0) {
-            return null;
-        } else {
-            return getFirstTopLevelBodyGeoPoint(formDef, formDef.getMainInstance());
-        }
-    }
-
-    /**
-     * Returns the reference of the first child of the given element that is of type geopoint and
-     * is not contained in a repeat.
-     */
-    private static TreeReference getFirstTopLevelBodyGeoPoint(IFormElement element, FormInstance primaryInstance) {
-        if (element instanceof QuestionDef) {
-            QuestionDef question = (QuestionDef) element;
-            int dataType = primaryInstance.resolveReference((TreeReference) element.getBind().getReference()).getDataType();
-
-            if (dataType == Constants.DATATYPE_GEOPOINT) {
-                return (TreeReference) question.getBind().getReference();
-            }
-        } else if (element instanceof FormDef || element instanceof GroupDef) {
-            if (element instanceof GroupDef && ((GroupDef) element).getRepeat()) {
-                return null;
-            } else {
-                for (IFormElement child : element.getChildren()) {
-                    // perform recursive depth-first search
-                    TreeReference geoRef = getFirstTopLevelBodyGeoPoint(child, primaryInstance);
-                    if (geoRef != null) {
-                        return geoRef;
-                    }
-                }
-            }
-        }
-
-        return null;
-    }
-
-    /**
-     * Returns the XPath path for the first geopoint in the primary instance that is before the given
-     * reference and not in a repeat.
-     */
-    private static String getInstanceGeoPointBefore(TreeReference firstBodyGeoPoint, TreeElement element) {
-        if (element.getRef().equals(firstBodyGeoPoint)) {
-            return null;
```

Comment:
@lognaturel, do you remember in which scenario this null should be returned? I'm not entirely clear on it, and when I remove it, the tests pass. I attempted to satisfy this if statement by editing the test, but to no avail. Given this, I didn’t implement it in the new parser, but if you can explain it, I might need to address it.

Reply:
Addressed in review comments.